app/controllers/posts_controller.rb

```diff
@@ -1,20 +1,22 @@
 class PostsController < ApplicationController
-  def new
-    @post = Post.new
-  end
-
-  def create
-    @post = Post.create(post_params)
-    redirect_to posts_url
-  end
 
   def index
-    @posts = Post.all
+    # this method is going to need parameters passed to it
+    @new_post = Post.new
+    @user_id = session[:user_id]
+    @list_posts = Post.all.order(created_at: :desc)
   end
 
+  def create
+    @new_post = Post.create(post_params)
+    redirect_to '/posts'
```

Comment:
would it be good to implement an if statement that reassured that its properly saved like in the user class?

Reply:
I agree with this, it's good to have a catch for when the `@new_post` is invalid so we shouldn't redirect to `/posts` in this scenario but instead just re-render the new post route. This is done for e.g. in the UsersController.

At the moment I think the only thing that could feasibly make a new post invalid was if the user id was invalid (which should in theory not happen given line 6) so it's not a huge deal, but if you did eventually make a migration to change the "message" column to be not null then I think it would be better than redirecting to `/posts` 

Also, I'd recommend using the variable `posts_url` incase, for whatever reason we wanted to change the naming of the posts route, for example maybe to `/feed` or something - then this would update automatically and not break everything

Reply:
Although it just occurred to me I can't actually remember if the posts page and the new posts page are different, you only actually have one view in posts which suggests they're the same? So maybe this isn't as big of a deal 🤷‍♀️

Reply:
Deleted my comment pls forget it if you read it already